src/gui/autoexpandabledialog.cpp

```diff
@@ -57,6 +57,13 @@ QString AutoExpandableDialog::getText(QWidget *parent, const QString &title, con
     d.m_ui->textEdit->setEchoMode(mode);
     d.m_ui->textEdit->setInputMethodHints(inputMethodHints);
 
+    const int lastDotIndex = text.lastIndexOf('.');
+    // select file name without extension, except 'extension'-only files like .gitignore
+    if (excludeExtension && (lastDotIndex > 0))
+        d.m_ui->textEdit->setSelection(0, lastDotIndex);
+    else
+        d.m_ui->textEdit->selectAll();
```

Comment:
maybe as such? to avoid invoking `lastIndexOf()` when not needed:
```c++
d.m_ui->textEdit->selectAll();
if (excludeExtension) {
    const int lastDotIndex = text.lastIndexOf('.');
    if (lastDotIndex > 0)
        d.m_ui->textEdit->setSelection(0, lastDotIndex);
}
```

Reply:
Then nothing will be selected when `(excludeExtension = true) && (text = ".gitignore")`